docs/howto/how-to-make-api-tool/api-tool-guide.md

```diff
@@ -0,0 +1,250 @@
+```python
+    def get_postseason_stats(player_id=237, season=2016):
+        resp = requests.get(
+            "https://www.balldontlie.io/api/v1/stats",
+            params={
+                "player_ids[]": player_id,
+                "postseason": True,
+                "seasons[]": season,
+            },
+        )
+        return resp.json()["data"]
+```
+From calling this function with `player_id=237` and `season=2016`, you will get back the data as a JSON.
+After returning the JSON data, we're able to read that into a [`pyarrow.Table`](https://arrow.apache.org/docs/python/generated/pyarrow.Table.html)
+
+### 2. Creating a `pyarrow.Table`
+Since we read and write data in the [Apache Arrow](https://arrow.apache.org/) data format, we will now convert this JSON to `pyarrow.Table` format with the following function:
+
+```python
+    def get_pyarrow_table_from_stats(stats_json):
+        stats_schema = pa.schema([
+            pa.field("pts", pa.int64()),
+            pa.field("reb", pa.int64()),
+            pa.field("ast", pa.int64()),
+            pa.field("stl", pa.int64()),
+            pa.field("blk", pa.int64()),
+            pa.field("min", pa.duration('s')),
+            pa.field("fga", pa.int64()),
+            pa.field("fgm", pa.int64()),
+            pa.field("fg3", pa.int64()),
+            pa.field("fta", pa.int64()),
+            pa.field("ftm", pa.int64()),
+            pa.field("turnovers", pa.int64()),
+        ])
+        assists = []
+        blocks = []
+        rebounds = []
+        minutes = []
+        points = []
+        steals = []
+        fga = []
+        fgm = []
+        fg3 = []
+        fta = []
+        ftm = []
+        tos = []
+        for stat in stats_json:
+            assists.append(stat["ast"])
+            blocks.append(stat["blk"])
+            rebounds.append(stat["reb"])
+            mins, secs = stat["min"].split(":")
+            mins_in_secs = int(mins)*60 + int(secs)
+            minutes.append(mins_in_secs)
+            points.append(stat["pts"])
+            steals.append(stat["stl"])
+            fga.append(stat["fga"])
+            fgm.append(stat["fgm"])
+            fg3.append(stat["fg3m"])
+            fta.append(stat["fta"])
+            ftm.append(stat["ftm"])
+            tos.append(stat["turnover"])
+        
+        table = pa.table(
+            [points, rebounds, assists, steals, blocks, minutes, fga, fgm, fg3, fta, ftm, tos],
+            schema=stats_chema
+        )
+        return table
+```
+This function iterates over the returned JSON from step 1 and pulls the relevant fields into an array of arrays, which conforms to the [`pa.schema`](https://arrow.apache.org/docs/python/generated/pyarrow.Schema.html) that we defined in the beginning of the function.
+
+> Other times, the data comes in a better format and you can convert to Apache Arrow format more easily with the [built-in helper function](https://arrow.apache.org/docs/python/json.html) 
```

Comment:
TW review - update to:

At times, data returns in a better format and you can convert it to Apache Arrow format with the [built-in helper function](https://arrow.apache.org/docs/python/json.html).

Reply:
i want to change this whole sentence actually. what about 

"In this example/For our example, the JSON that is returned from the API cannot be automatically parsed. Generally, we can avoid manual conversion to the Apache Arrow format with the [built-in helper function](https://arrow.apache.org/docs/python/json.html)."

Reply:
@ayx-orion sounds good, just a minor edit:

"In this example, the JSON that is returned from the API can't be automatically parsed. Generally, we can avoid manual conversion to the Apache Arrow format with the [built-in helper function](https://arrow.apache.org/docs/python/json.html)."